syntax="proto3";
package order;


service Order{
    rpc SendOrder(OrderMessage) returns (MessageReply) {}
    rpc TakeStock(OrderStatus) returns (MessageReply) {}
}

message OrderMessage {
  int32 customer_id=1;
  string action=2;
  string customer_email =3;
  repeated Item items = 4;
}

message Item {
  int32 item_id=1;
  int32 item_qty=2;
  float cost = 3;
}

message MessageReply {
  int32 id=1;
  string orderStatus=2;
  repeated ItemStatus itemResp = 3;
}


message ItemStatus {
  int32 itemId =1;
  string currentState =2;
}

message OrderStatus {
  int32 orderId =1;
  string customer_email =2;
}

message OrderStatusResp {
  int32 orderId =1;
  string customer_email =2;
  string orderStatus=3;
  repeated ItemStatus itemResp = 4;
}









